Sources/SwiftDiscord/Rest/DiscordRateLimiter.swift

```diff
@@ -152,30 +152,65 @@ public final class DiscordRateLimiter {
 /// Ex. /channels/232184444340011009/messages and /channels/186926276592795659/messages
 /// Are considered different endpoints
 public struct DiscordRateLimitKey: Hashable {
+	/// URL Parts for the purpose of rate limiting.
+    /// Combine all the parts of the URL into a list of which parts exist
+    /// Ex. /channels/232184444340011009/messages would be represented by [.channels, .channelID, .messages]
+    public struct DiscordRateLimitURLParts: OptionSet {
```

Comment:
Yes, but no more than the use of an enum for endpoint limits the number of possible endpoints.  If you're worried about the number of different url parts, if we go over 32, we can switch to a `UInt64`, and if we go over 64 (which I doubt will happen, there's only so many things that would make sense to put in a URL), hopefully [this](https://github.com/apple/swift-evolution/blob/master/proposals/0104-improved-integers.md) will have gone through and we can just use a `DoubleWidth<UInt64>`, but if not, we can make a custom struct for the purpose.